### PR TITLE
bind kong admin to local expose less ports

### PR DESCRIPTION
**Pull request by hutchic**
Kong admin api should be bound to 127 only as a sensible default. While I was in the area exposing redis, postgres, consul and vault at all seems fraught for mistakes

### CODE REVIEW

**Comment by bnevis-i** [comment]
@hutchic I was expecting localhost binding for all of the exposed ports, not wholesale removal of exposed ports.  For example:

```
      - "8400:8400"	
      - "8500:8500"
```
becomes
```
      - "127.0.0.1:8400:8400"	
      - "127.0.0.1:8500:8500"
```

I support to removal of the port binding to the host's external ports, but I think this request goes too far because a lot of developers start parts of the framework with docker, and then develop applications on top not in a container and expect localhost:port to work.   I, for instance, often find myself hitting the consul web UI and removal of the consul ports would be a major inconvenience.

**Comment by lenny-intel** [comment]
In addition to @bnevis-i comment. These changes should only be in the nightlybuild (not fuji) and only in the secure compose files, not the no-sec files.

File: releases/fuji/compose-files/docker-compose-fuji-no-secty.yml [hunk 1]
```diff
@@ -49,9 +49,6 @@ services:
 
   consul:
     image: consul:1.3.1
-    ports:
-      - "8400:8400"
-      - "8500:8500"
```
Comment (jim-wang-intel) [inline releases/fuji/compose-files/docker-compose-fuji-no-secty.yml]: Should we retro-change the Fuji version? Maybe the decision should be called for attention by the groups?

File: releases/fuji/compose-files/docker-compose-fuji.yml [hunk 2]
```diff
@@ -51,9 +51,6 @@ services:
 
   consul:
     image: edgexfoundry/docker-edgex-consul:1.1.0
-    ports:
-      - "8400:8400"
-      - "8500:8500"
```
Comment (jim-wang-intel) [inline releases/fuji/compose-files/docker-compose-fuji.yml]: Ditto question about the retro-change.

File: releases/fuji/compose-files/docker-compose-redis-fuji-no-secty.yml [hunk 3]
```diff
@@ -49,9 +49,6 @@ services:
 
   consul:
     image: consul:1.3.1
-    ports:
-      - "8400:8400"
-      - "8500:8500"
```
Comment (jim-wang-intel) [inline releases/fuji/compose-files/docker-compose-redis-fuji-no-secty.yml]: Ditto retro-change

File: releases/nightly-build/compose-files/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1 @@
+docker-compose-nexus-mongo.yml
```
Comment (jim-wang-intel) [inline releases/nightly-build/compose-files/docker-compose.yml]: I like this composite way to select which version of compose file to use! 👍

Comment (bnevis-i) [reply releases/nightly-build/compose-files/docker-compose.yml]: This probably shouldn't have been included in the PR.

File: releases/fuji/compose-files/docker-compose-fuji-no-secty.yml (hunk 1 above)
Comment (lenny-intel) [reply releases/fuji/compose-files/docker-compose-fuji-no-secty.yml]: If we did this it would have to be a point release, which I don't think it warrants as of security hardening is evolving release to release, IMHO. @jpwhitemn thoughts?

File: releases/nightly-build/compose-files/docker-compose.yml (hunk 4 above)
Comment (jim-wang-intel) [reply releases/nightly-build/compose-files/docker-compose.yml]: I vote for this way :-)

Comment (lenny-intel) [reply releases/nightly-build/compose-files/docker-compose.yml]: Like it, but yes it should be in a separate PR.

Comment (jim-wang-intel) [reply releases/nightly-build/compose-files/docker-compose.yml]: got cha

Comment (hutchic) [reply releases/nightly-build/compose-files/docker-compose.yml]: whoops sorry. I did a lazy `git add .` and missed that it grabbed this :)

Comment (bnevis-i) [reply releases/nightly-build/compose-files/docker-compose.yml]: We like this so much we thought it would be a great idea if you added docker-compose.yml to the .gitignore.

**Comment by hutchic** [comment]
@bnevis-i yup as discussed at the security working group although this repository is called "developer-scripts" it seems implied that this is production worthy docker-compose. I think the conclusion was in the short term bind to 127 and then either make the hardening / production guide more prominent or take the approach here and drop the ports entirely.

TLDR; I'm going to bind to 127 in this PR. I may open separate PR's to add a "DEVELOPMENT" flag which exposes the datastores via Kong

**Comment by hutchic** [comment]
@jim-wang-intel I wasn't sure if this would be considered a "security fix" and therefore worth backporting to the current release as well as nightly

File: releases/fuji/compose-files/docker-compose-fuji-no-secty.yml (hunk 1 above)
Comment (hutchic) [reply releases/fuji/compose-files/docker-compose-fuji-no-secty.yml]: @jim-wang-intel I wasn't sure if this would be considered a "security fix" and therefore worth backporting to the current release (though current release is Geneva isn't it 😬  ) as well as nightly

**Comment by hutchic** [comment]
@lenny-intel I humbly disagree re: don't apply the changes to the no-sec files

I think the `no-sec` files should be bound to 127 by default (could do a ENV for override if warrented). The security enabled should have a very nominal amount of ports exposed at all but the ability to set a `DEVELOPMENT_MODE` flag that makes them accessible through Kong -- is what my train of thought has adjusted to :)

**Comment by lenny-intel** [comment]
> I think the `no-sec` files should be bound to 127 by default (could do a ENV for override if warrented).

I am ok with this as it doesn't impact dev running on the same system.

**Comment by bnevis-i** [comment]
Intending to look at this... but will be delayed due to getting ready for the Hanoi prewire.

**Comment by bnevis-i** [comment]
@hutchic Lots of docker-compose changes last few days.  Suggest abandoning this PR and doing this afresh as a single commit based off the current master.
